python/lsst/cp/testing/isr.py

```diff
@@ -0,0 +1,329 @@
+        if not config.doBrighterFatter:
+            del self.bfKernel
+            del self.newBFKernel
+        if not config.doDefect:
+            del self.defects
+        if not config.doDark:
+            del self.dark
+        if not config.doFlat:
+            del self.flat
+        if not config.doFringe:
+            del self.fringes
+        if not config.doStrayLight:
+            del self.strayLightData
+        if not config.usePtcGains and not config.usePtcReadNoise:
+            del self.ptc
+        if not config.doAttachTransmissionCurve:
+            del self.opticsTransmission
+            del self.filterTransmission
+            del self.sensorTransmission
+            del self.atmosphereTransmission
+        else:
+            if not config.doUseOpticsTransmission:
+                del self.opticsTransmission
+            if not config.doUseFilterTransmission:
+                del self.filterTransmission
+            if not config.doUseSensorTransmission:
+                del self.sensorTransmission
+            if not config.doUseAtmosphereTransmission:
+                del self.atmosphereTransmission
+        if not config.doIlluminationCorrection:
+            del self.illumMaskedImage
+
+        if not config.doWrite:
+            del self.outputExposure
+            del self.preInterpExposure
+            del self.outputFlattenedThumbnail
+            del self.outputOssThumbnail
+        if not config.doSaveInterpPixels:
+            if "preInterpExposure" in self.outputs:
+                del self.preInterpExposure
+        if not config.qa.doThumbnailOss:
+            if "outputOssThumbnail" in self.outputs:
+                del self.outputOssThumbnail
+        if not config.qa.doThumbnailFlattened:
+            if "outputFlattenedThumbnail" in self.outputs:
+                del self.outputFlattenedThumbnail
+
+    def adjustQuantum(self, inputs, outputs, label, data_id):
+        """Adjust quantum to drop task execution depending on the input data.
+
+        Parameters
+        ----------
+        inputs : `dict`
+            Dictionary of input connections.
+        outputs : `Mapping`
+            Mapping of output datasets.
+        label : `str`
+            Task label.
+        data_id : `lsst.daf.butler.DataCoordinate`
+            Data id for this task execution.
+
+        Returns
+        -------
+        adjustedInputs : `Mapping`
+            Adjusted set of inputs.
+        adjustedOutputs : `Mapping`
+            Adjusted set of outputs.
+
+        Raises
+        ------
+        NoWorkFound
+            Raised if the task execution should be skipped.
+        """
+        inputConnection, inputExpRefs = inputs['ccdExposure']
+        inputExpRef = inputExpRefs[0]
+        doWork = True
+
+        if self.config.expectedExposureType != "":
+            expected = self.config.expectedExposureType.lower()
+            headerValue = inputExpRef.dataId.exposure.observation_type
```

Comment:
I don't like calling this `headerValue` since it's not from the header, it's from the Butler.  Why not `observationType`?